### PR TITLE
Refactor linting + test settings to support passing IntegrationTest or other Configurations in

### PR DESCRIPTION
**Pull request by ericdotdata**
Example from ci test:

```
lazy val root = project
  .in(file("."))
  .configs(IntegrationTest)
  .settings(Defaults.itSettings)
  .settings(dotDataSettings(testConfigurations = Seq(Test, IntegrationTest)))
```

This also always uses the `Compile / resourceManaged` value regardless of the configuration for the linting/code formatting.

### CODE REVIEW

**Comment by ericdotdata** [comment]
Need a tweak still:

```
[error] /Users/eric/dd/bigdata/core_artifacts_format_upgrader_spark_app/src/it/scala/core_artifacts_format_upgrader_spark_app/SparkWebUtilSpec.scala:26:56: discarded non-Unit value
[error]   private def createHdfsTestSpace(): Unit = hdfs.mkdirs(new Path(hdfsTestSpace))
[error]                                                        ^
[error] /Users/eric/dd/bigdata/core_artifacts_format_upgrader_spark_app/src/it/scala/core_artifacts_format_upgrader_spark_app/SparkWebUtilSpec.scala:33:56: discarded non-Unit value
[error]   private def deleteHdfsTestSpace(): Unit = hdfs.delete(new Path(hdfsTestSpace), true)
[error]                                                        ^
[error] two errors found
[error] (IntegrationTest / compileIncremental) Compilation failed
[error] Total time: 12 s, completed Sep 21, 2021 11:22:55 AM
```

**Comment by ericdotdata** [comment]
Next up:

```
[info] compiling 10 Scala sources to /Users/eric/dd/bigdata/yarn_logs_provider/target/scala-2.12/classes ...
[error] /Users/eric/dd/bigdata/yarn_logs_provider/src/main/scala/logsprovider/YarnLogsProviderServer.scala:75:31: method toApplicationId in class ConverterUtils is deprecated
[error]           Some(ConverterUtils.toApplicationId(yarnAppId))
[error]                               ^
[error] one error found
[error] (Compile / compileIncremental) Compilation failed
[error] Total time: 3 s, completed Sep 21, 2021 1:17:11 PM
```

**Comment by er1c** [comment]
@patrickgalbreath @dot-vlad this is ready to review

**Comment by dot-vlad** [comment]
Hey, sorry, I don't understand what is the problem we're trying to solve here. There is quite a lot of code changes, and it is hard to infer what was the original problem, and I don't see a connection with any of the existing themes and I don't see what's the problem with "this also always uses the Compile / resourceManaged value regardless of the configuration for the linting/code formatting." and looks like IntegrationTest configuration is already passing.

**Comment by ericdotdata** [comment]
@dot-vlad its not passing, because the current `bigdata` doesn't actually enforce fatal warnings - the latest plugin version actually fixed this, but isnt applied yet since I was trying to find other problems that needed to be filtered out from the fatal warnings method.

**Comment by ericdotdata** [comment]
@dot-vlad ahh here we go: https://github.com/ramencloud/bigdata/pull/27242 you can see the corresponding changes for various projects for this update

**Comment by dot-vlad** [comment]
@ericdotdata 

> its not passing

What is _it_ ?

**Comment by ericdotdata** [comment]
@dot-vlad https://github.com/ramencloud/sbt-config/commit/6ef20ec87db577528f563b12353b33c831d4ffe4#diff-bf0116b11452dd8fb7779888f127e0f11399b20e4b551ce9b1b756c2f5fbccdfL158 this wasn't being used, so if we update bigdata to use the current fixed sbt-config you get failures: https://github.com/ramencloud/bigdata/pull/27438

File: src/sbt-test/linting/fail-unused-import-test/test [hunk 1]
```diff
@@ -1,2 +1,5 @@
 > compile
--> Test / compile
```
Comment (analizator1) [inline src/sbt-test/linting/fail-unused-import-test/test:2]: Just a hint for newbies on syntax of this file: https://www.scala-sbt.org/1.x/docs/Testing-sbt-plugins.html.

**Comment by dot-vlad** [comment]
> @dot-vlad [6ef20ec#diff-bf0116b11452dd8fb7779888f127e0f11399b20e4b551ce9b1b756c2f5fbccdfL158](https://github.com/ramencloud/sbt-config/commit/6ef20ec87db577528f563b12353b33c831d4ffe4#diff-bf0116b11452dd8fb7779888f127e0f11399b20e4b551ce9b1b756c2f5fbccdfL158) this wasn't being used, so if we update bigdata to use the current fixed sbt-config you get failures: [ramencloud/bigdata#27438](https://github.com/ramencloud/bigdata/pull/27438)
> 
> This helps mitigate some of those failures and enable the syntax in the above linked PR

What it became a problem, we updated the version of the plugin a number of times before?

I am still not getting what this PR is trying to achieve.

**Review by kondziolka9ld — APPROVED** [review]

**Comment by analizator1** [comment]
> Guys, this part: "test settings to support passing IntegrationTest" still doesn't work in web-api

Fix in https://github.com/ramencloud/sbt-config/pull/54.

**Pull request merged**